### PR TITLE
Fix incorrect .NET error

### PR DESCRIPTION
**Pull request by SakshiS-harma**
This PR provides a fix for #16676.
The error condition when dotnet couldn't be found (by running dotnet --version) didn't provide appropriate results, the code should default to present the error that states .NET core isn't installed; but it was instead presenting the installed .NET core version isn't supported- which is incorrect. This is fixed. 
Also, while throwing installation or unsupported error for .NET core, the build failure presented impractical message- this is fixed as well.
image
The highlighted part doesn't show up because there is nothing in the output pane to look for. As the prior behavior, it still shows the same error twice- once as an error and another to collect data from user (with options).
image

### CODE REVIEW

File: extensions/sql-database-projects/src/tools/netcoreTool.ts [hunk 1]
```diff
@@ -148,15 +148,21 @@ export class NetCoreTool {
...
-						isSupported = false;
+					try {
+						if (semver.gte(this.netCoreSdkInstalledVersion, minSupportedNetCoreVersion)) {		// Net core version greater than or equal to minSupportedNetCoreVersion are supported for Build
+							isSupported = true;
+						} else {
+							isSupported = false;
+						}
+						resolve({ stdout: this.netCoreSdkInstalledVersion });
+					} catch (err) {
+						console.log(err);
```
Comment (Charles-Gagnon) [inline extensions/sql-database-projects/src/tools/netcoreTool.ts:159]: Do you need to update the netCoreInstallState if this happens?

File: extensions/sql-database-projects/src/tools/netcoreTool.ts [hunk 2]
```diff
@@ -246,3 +252,9 @@ export class NetCoreTool {
 			});
 	}
 }
+
+export class DotNetError extends Error {
+	constructor(message?: string) {
```
Comment (Charles-Gagnon) [inline extensions/sql-database-projects/src/tools/netcoreTool.ts:257]: You shouldn't need this constructor

**Review by Charles-Gagnon — APPROVED** [review]

File: extensions/sql-database-projects/src/tools/netcoreTool.ts (hunk 1 above)
Comment (SakshiS-harma) [reply extensions/sql-database-projects/src/tools/netcoreTool.ts:159]: This catch was actually moving on to reach the outer catch statement, where netCoreInstallState gets set, hence doesn't need another redundant update here.

File: extensions/sql-database-projects/src/tools/netcoreTool.ts (hunk 2 above)
Comment (Benjin) [reply extensions/sql-database-projects/src/tools/netcoreTool.ts:257]: Whoops; my mistake.  Charles is correct.

**Review by Benjin — APPROVED** [review]

File: extensions/sql-database-projects/src/tools/netcoreTool.ts (hunk 2 above)
Comment (SakshiS-harma) [reply extensions/sql-database-projects/src/tools/netcoreTool.ts]: Done.

**Pull request merged**